core/data/src/main/java/uk/gov/gchq/gaffer/data/generator/OpenCypherCsvGenerator.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.data.generator;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.apache.commons.lang3.StringUtils;
+
+import uk.gov.gchq.gaffer.data.element.Element;
+import uk.gov.gchq.gaffer.data.element.IdentifierType;
+import uk.gov.gchq.koryphe.Since;
+import uk.gov.gchq.koryphe.Summary;
+
+import static uk.gov.gchq.gaffer.data.generator.OpenCypherCsvElementGenerator.DESTINATION;
+import static uk.gov.gchq.gaffer.data.generator.OpenCypherCsvElementGenerator.EDGE_GROUP;
+import static uk.gov.gchq.gaffer.data.generator.OpenCypherCsvElementGenerator.ENTITY_GROUP;
+import static uk.gov.gchq.gaffer.data.generator.OpenCypherCsvElementGenerator.NEO4J_DESTINATION;
+import static uk.gov.gchq.gaffer.data.generator.OpenCypherCsvElementGenerator.NEO4J_EDGE_GROUP;
+import static uk.gov.gchq.gaffer.data.generator.OpenCypherCsvElementGenerator.NEO4J_ENTITY_GROUP;
+import static uk.gov.gchq.gaffer.data.generator.OpenCypherCsvElementGenerator.NEO4J_SOURCE;
+import static uk.gov.gchq.gaffer.data.generator.OpenCypherCsvElementGenerator.NEO4J_VERTEX;
+import static uk.gov.gchq.gaffer.data.generator.OpenCypherCsvElementGenerator.SOURCE;
+import static uk.gov.gchq.gaffer.data.generator.OpenCypherCsvElementGenerator.VERTEX;
+
+import java.util.Collection;
+import java.util.LinkedHashMap;
+
+/**
+ * Generates an openCypher formatted CSV string for each {@link Element}. An Elements Vertex, Source, Destination
+ * and Group are mapped to their openCypher equivalents, either AWS Neptune or Neo4j depending on whether the flag is set.
+ * An Elements properties and their respective types are taken from the schema.
+ */
+@Since("2.0.0")
+@Summary("Generates an openCypher formatted CSV string for each element")
+public class OpenCypherCsvGenerator extends CsvGenerator {
+    private LinkedHashMap<String, String> fields = new LinkedHashMap<>();
+
+    /**
+     * When set to true, the headers will be formatted to match those used by neo4j.
+     */
+    private boolean neo4jFormat = false;
+
+    @Override
+    public String _apply(final Element element) {
+        final StringBuilder strBuilder = new StringBuilder();
+        for (final String field : fields.keySet()) {
+            final Object value = getFieldValue(element, field);
+
+            if (null != value) {
```

Comment:
Can we use value != null. Yoda conditions aren't best practice in Java, especially as we aren't expecting null here.